### PR TITLE
feat: updated rate/routing tooltips

### PR DESCRIPTION
**Pull request by cartcrom**
## Description

Updates Exchange rate & routing line items/tooltips to new UI design

## Screen capture

|    Before    |   After    |
| ------------ | ------------ |
| Screenshot 2023-10-04 at 3 41 55 PM | Screenshot 2023-10-04 at 3 19 56 PM |

|    Before    |   After   |
| ------------ | ----------- |
| image  | image |

### CODE REVIEW

File: src/components/swap/SwapRoute.tsx [hunk 1]
```diff
@@ -1,64 +1,83 @@
...
-  const { chainId } = useWeb3React()
-  const autoRouterSupported = useAutoRouterSupported()
+// TODO(WEB-2022)
+// Can `trade.gasUseEstimateUSD` be defined when `chainId` is not in `SUPPORTED_GAS_ESTIMATE_CHAIN_IDS`?
+function getGasPrice(trade: ClassicTrade) {
+  const chainId = trade.inputAmount.currency.chainId
+  if (!trade.gasUseEstimateUSD || !SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)) return undefined
 
-  const routes = getRoutingDiagramEntries(trade)
+  return trade.gasUseEstimateUSD === 0 ? '<$0.01' : '$' + trade.gasUseEstimateUSD.toFixed(2)
```
Comment (JackShort) [inline src/components/swap/SwapRoute.tsx:22]: you should just return a number here and let a formatter do the formatting after its returned

File: src/components/swap/SwapRoute.tsx [hunk 2]
```diff
@@ -1,64 +1,83 @@
...
-      )}
+      <UniswapXDescription />
+    </Column>
+  )
+}
+
+export function SwapRoute({ trade }: { trade: ClassicTrade }) {
+  const { inputAmount, outputAmount } = trade
+  const routes = getRoutingDiagramEntries(trade)
+  const gasPrice = getGasPrice(trade)
```
Comment (JackShort) [inline src/components/swap/SwapRoute.tsx:66]: you can do the following to get the same formatting if you return a number in the `getGasPrice` function:

```
const gasPrice = useFormatter().formatNumber({ input: getGasePrice(trade), type: [
  { exact: 0, hardCodedInput: { input: 0.01, prefix: '<' }, formatterOptions: {
    notation: 'standard',
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
  }, {
    upperBound: Infinity, formatterOptions: {
    notation: 'standard',
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
    },
  }
] 
})
```

**Review by JackShort — APPROVED** [review]
fine with me i can add the formatting afterwards in a sep pr that touches the swap modal if you wouldlike

File: src/components/swap/SwapRoute.tsx (hunk 2 above)
Comment (cartcrom) [reply src/components/swap/SwapRoute.tsx:66]: wow that looks terrifying. yeah I think updating NumberType.FiatGasPrice to include the exact zero case in sep PR probably makes sense

File: src/components/swap/SwapRoute.tsx [hunk 3]
```diff
@@ -1,64 +1,83 @@
...
+export function SwapRoute({ trade }: { trade: ClassicTrade }) {
+  const { inputAmount, outputAmount } = trade
+  const routes = getRoutingDiagramEntries(trade)
+  const gasPrice = useGasPrice(trade)
+
+  return useAutoRouterSupported() ? (
+    <Column gap="md">
+      <RoutingDiagram routes={routes} currencyIn={inputAmount.currency} currencyOut={outputAmount.currency} />
+      <ThemedText.Caption color="neutral2">
+        {gasPrice && <Trans>Best price route costs ~{gasPrice} in gas. </Trans>}
```
Comment (natew) [inline src/components/swap/SwapRoute.tsx:72]:
```suggestion
        {Boolean(gasPrice) && <Trans>Best price route costs ~{gasPrice} in gas. </Trans>}
```

idk if this is a number type but if so this is important

**Review by natew — APPROVED** [review]

**Comment by natew** [comment]
tests/types need fixing also

**Review by JackShort — APPROVED** [review]

**Pull request merged**